File: lib/domain/entities/book.dart
class Book {
  final String id;
  final String title;
  final List<String> authors;
  final String description;
  final String thumbnailUrl;
  final String coverLocalPath;
  final List<String> categories;
  final int pageCount;
  final String language;
  final String previewLink;
  final String webReaderLink;
  final String source;
  final String pdfDownloadLink;
  final String epubDownloadLink;
  final String localFilePath;
  final bool isDownloaded;

  const Book({
    required this.id,
    required this.title,
    required this.authors,
    required this.description,
    required this.thumbnailUrl,
    this.coverLocalPath = '',
    required this.categories,
    required this.pageCount,
    required this.language,
    required this.previewLink,
    required this.webReaderLink,
    required this.source,
    this.pdfDownloadLink = '',
    this.epubDownloadLink = '',
    this.localFilePath = '',
    this.isDownloaded = false,
  });
}

File: backend/BookReader.Api/Entities/Book.cs
using System;
using System.Collections.Generic;

namespace BookReader.Api.Entities;

public partial class Book
{
    public int Id { get; set; }

    public string? GoogleBookId { get; set; }

    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    public string? ThumbnailUrl { get; set; }

    public string? CoverLocalPath { get; set; }

    public int PageCount { get; set; }

    public string? Language { get; set; }

    public string? PreviewLink { get; set; }

    public string? WebReaderLink { get; set; }

    public string Source { get; set; } = null!;

    public string? PdfDownloadLink { get; set; }

    public string? EpubDownloadLink { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual ICollection<Bookmark> Bookmarks { get; set; } = new List<Bookmark>();

    public virtual ICollection<NoteHighlight> NoteHighlights { get; set; } = new List<NoteHighlight>();

    public virtual ICollection<ReadingProgress> ReadingProgresses { get; set; } = new List<ReadingProgress>();

    public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();

    public virtual ICollection<UserLibrary> UserLibraries { get; set; } = new List<UserLibrary>();

    public virtual ICollection<Author> Authors { get; set; } = new List<Author>();

    public virtual ICollection<Category> Categories { get; set; } = new List<Category>();
}
